src/mgmt/cli_mgmt.c

```diff
@@ -943,12 +952,16 @@ sys_attach(const char *name, struct dc_mgmt_sys **sysp)
  * \param[in,out]	sys	system handle
  */
 int
-dc_mgmt_sys_attach(const char *name, struct dc_mgmt_sys **sysp)
+dc_mgmt_sys_attach(const char *name, bool use_global_resp, struct dc_mgmt_sys **sysp)
 {
 	if (name == NULL)
 		name = DAOS_DEFAULT_SYS_NAME;
 
-	return sys_attach(name, sysp);
+	/** only use the global attach response if default system (force to re-attach if not) */
+	if (strcmp(name, DAOS_DEFAULT_SYS_NAME) != 0)
+		use_global_resp = false;
+
+	return sys_attach(name, use_global_resp, sysp);
```

Comment:
[Question] If a system's name is not "daos_server", this `if` statement would be wrong?

Reply:
im not sure what you mean by wrong. in any cae, i would like to land the other PR that allows a different sys name than daos_server first, and this PR needs to be adjusted anyway.

Reply:
OK I believe this should be fixed now.